app/helpers/automated_tests_helper.rb

```diff
@@ -123,62 +121,120 @@ def create_ant_test_files(assignment)
   def process_test_form(assignment, params)
 
     # Hash for storing new and updated test files
-    updated_files = {}
-
-    # Retrieve all test file entries
-    testfiles = params[:test_files_attributes]
-
-    # First check for duplicate filenames:
-    filename_array = []
-    testfiles.values.each do |tfile|
-      if tfile['filename'].respond_to?(:original_filename)
-        fname = tfile['filename'].original_filename
-        # If this is a duplicate filename, raise error and return
-        if filename_array.include?(fname)
-          raise I18n.t('automated_tests.duplicate_filename') + fname
-        else
-          filename_array << fname
+    updated_script_files = {}
+    updated_support_files = {}
+
+    # Array for checking duplicate file names
+    file_name_array = []
+
+    # add existing scripts names
+    params.each { |key, value| if(!key[/test_script_\d+/].nil?) then
+                                 file_name_array << value end }
```

Comment:
`end` at 132, 58 is not aligned with `if` at 131, 31.<br>Expression at 132, 63 should be on its own line.